RPIPs/RPIP-44.md

```diff
@@ -18,16 +18,36 @@ This proposal specifies when [execution layer triggerable exits as defined in EI
-  - This function MAY be called by any user if the Node has a `deficit` greater than `exit_deficit`
-    - It MUST NOT be possible to exit more than the minimum number of validators needed to cover `deficit`
-- `debt` is incurred via penalties, as enumerated in [RPIP-42](RPIP-42.md)
-- The protocol SHALL use `rewards` and/or `credit` held by the NO to decrease `debt` by the same amount prior to taking action on exits
-- The protocol SHOULD use staked RPL held by the NO to decrease `debt` by the corresponding amount prior to taking action on exits
-- The initial settings SHALL be:
-  - `exit_deficit`: 0.2 ETH
+
+This specification extends the specification of megapools in
+[RPIP-43](RPIP-43.md) with a facility for removal of validators from the
+megapool to be done by parties other than the Node Operator.
+
+Funds (ETH) associated with a megapool SHALL be accounted in (at least) the following categories:
+- Credits (`credit`): Funds that can be used towards validator deposits; sources include but are not limited to deposits from the Node Operator, ETH staked on behalf of the node ([RPIP-32](RPIP-32.md)), and ETH already staked on the beacon chain for a migrating validator
+- Deposits (`deposited`): Funds initially received as credits that are currently staked on the beacon chain for validators associated with this megapool
+- Withdrawals (`withdrawn`): Funds received into the megapool via withdrawals from the beacon chain, including both principal and the node's share of rewards, NOT including pool stakers' share of the rewards
+- Penalties (`debt`): Funds lost to penalties as enumerated in [RPIP-42](RPIP-42.md), or repaid from other balances as specified in this RPIP
+
+The following quantity, `deficit`, is derived based on the megapool's balances:
+```math
+$$ \mathtt{deficit} = \mathtt{credit} + \mathtt{withdrawn} - \mathtt{debt} $$
+```
+
+We also introduce a protocol-wide parameter `exit_deficit`, with initial setting
+```math
+$$ \mathtt{exit\_deficit} = 0.2\;\mathrm{ETH} $$
+```
+
+The exit functionality for a megapool is specified as follows:
+- A megapool contract SHALL include an `exit` function
+  - This function SHALL accept one or more public keys for validators associated with this megapool that have exited the beacon chain
+  - This function SHALL either remove the specified validators from the megapool or revert
+  - This function MAY be freely called by the Node Operator (as implied by [RPIP-43](RPIP-43.md))
+  - This function MAY be called by any account under the condition: `deficit >= exit_deficit`
+      - It MUST NOT be possible for accounts other than the Node Operator to exit more validators than needed to reduce `deficit` below `exit_deficit`
+- The protocol SHALL use the `withdrawn` and `credit` balances to decrease `debt` prior to taking action on exits
+- The protocol SHOULD use staked RPL to decrease `debt` by the corresponding amount prior to taking action on exits
```

Comment:
Hmmm... we should somehow account for credit coming from "exiting" validators. Ie, if 2 is the max validators that can be forced to exit, I shouldn't be able to exit multiple sets of 2. This could use the proposed pending_removal state, and set force-exited validators' rp_status to pending_removal.